airflow/include/tasks/extract/astronomer_providers_docs.py

```diff
@@ -1,14 +1,47 @@
 from __future__ import annotations
 
+import re
+
 import pandas as pd
+import urllib3
+from bs4 import BeautifulSoup
+from weaviate.util import generate_uuid5
 
-from include.tasks.extract.utils.html_utils import get_internal_links, urls_to_dataframe
+from include.tasks.extract.utils.html_utils import fetch_page_content, get_internal_links
 
 
 def extract_provider_docs() -> list[pd.DataFrame]:
     exclude_docs = ["_api", "_modules", "_sources", "changelog.html", "genindex.html", "py-modindex.html", "#"]
     base_url = "https://astronomer-providers.readthedocs.io/en/stable/"
 
-    urls = get_internal_links(base_url, exclude_docs)
-    df = urls_to_dataframe(urls, "astronomer-providers")
+    all_links = get_internal_links(base_url, exclude_docs)
+    # df = urls_to_dataframe(urls, "astronomer-providers")
+
+    docs_url_parts = urllib3.parse.urlsplit(base_url)
+    docs_url_base = f"{docs_url_parts.scheme}://{docs_url_parts.netloc}"
+
+    # make sure we didn't accidentally pickup any unrelated links in recursion
+    # get rid of older versions of the docs, only care about "stable" version docs
+    old_version_doc_pattern = r"/(\d+\.)*\d+/"
+    non_doc_links = {
+        link if (docs_url_base not in link) or re.search(old_version_doc_pattern, link) or "/latest/" in link else ""
```

Comment:
nitpick: should we make `(docs_url_base not in link) or re.search(old_version_doc_pattern, link) or "/latest/"` a function for better readability?